server/src/main/java/org/elasticsearch/cluster/metadata/MetadataIndexTemplateService.java

```diff
@@ -295,12 +320,41 @@ static ClusterState addIndexTemplateV2(final ClusterState currentState, final bo
             deprecationLogger.deprecated(warning);
         }
 
-        // TODO: validation of index template
```

Comment:
Maybe leave the todo? I think we still need to validate whether templates with the same priority exist that have overlapping patterns?

Reply:
This todo was for validating the mappings and settings in an index template (which this PR adds), we have a TODO item in the list on #53101 to make sure explicitly add the check for overlapping v2 template, so no need to worry!